sdk/communication/AzureCommunicationChat/Source/ChatClient.swift

```diff
@@ -376,49 +395,52 @@ public class ChatClient {
             )
 
             // Register for push notifications
-            registrarClient?.setRegistration(with: clientDescription, for: [transport]) { response, error in
-                if let error = error {
-                    completionHandler(
-                        .failure(AzureError.client("Failed to start push notifications", error)),
-                        response
-                    )
-                } else {
+            registrarClient.setRegistration(with: clientDescription, for: [transport]) { result in
+                switch result {
+                case .success(let response):
                     self.pushNotificationsStarted = true
-                    completionHandler(.success(()), response)
+                    completionHandler(.success(response))
+                case .failure(let error):
+                    self.options.logger.error("Failed to start push notifications with error: \(error.localizedDescription)")
+                    completionHandler(.failure(AzureError.client("Failed to start push notifications", error)))
                 }
             }
         } catch {
-            options.logger.error("Failed to start push notifications with error: \(error)")
+            options.logger.error("Failed to start push notifications with error: \(error.localizedDescription)")
+            completionHandler(.failure(AzureError.client("Invalid URL.", error)))
         }
     }
 
+
     /// Stop push notifications.
     /// - Parameter completionHandler: Success indicates push notifications have been stopped.
     public func stopPushNotifications(
-        completionHandler: @escaping (Result<Void, AzureError>, HTTPURLResponse?) -> Void
+        completionHandler: @escaping (Result<HTTPResponse?, AzureError>) -> Void
     ) {
         guard pushNotificationsStarted else {
-            completionHandler(.failure(AzureError.client("Push notifications are not enabled.")), nil)
+            completionHandler(.failure(AzureError.client("Push notifications are not enabled.")))
             return
         }
 
-        registrarClient?.deleteRegistration { response, error in
-            if let error = error {
-                completionHandler(.failure(AzureError.client("Failed to stop push notifications", error)), response)
-            } else {
+        registrarClient?.deleteRegistration { result in
```

Comment:
So with this, if there's no registrar client and someone causes `start/stopPushNotifications`, it will silently fail. Is that what you want? It would seem like it's not since you deliberately return an error if push notifications haven't started.

Reply:
Thanks! Already fixed and will push the change.